File: utils/thumbnail/check_pixels.py
import cv2
import sys

def on_mouse(event, x, y, flags, userdata):
    if event == cv2.EVENT_MOUSEMOVE:
        print(f"Mouse Position: ({x}, {y})")

def main():
    if len(sys.argv) < 2:
        print(f"Usage: python {sys.argv[0]} <image_path>")
        sys.exit(1)

    image_path = sys.argv[1]
    img = cv2.imread(image_path)

    if img is None:
        print("Failed to load image.")
        sys.exit(1)

    cv2.namedWindow("image", cv2.WINDOW_AUTOSIZE)
    cv2.setMouseCallback("image", on_mouse)

    while True:
        cv2.imshow("image", img)
        key = cv2.waitKey(10)
        if key == 27 or key == ord('q'):  # ESC or q
            break

    cv2.destroyAllWindows()

if __name__ == "__main__":
    main()

File: utils/thumbnail/check_pixels.cpp
#include <iostream>
#include <opencv2/opencv.hpp>

void onMouse(int event, int x, int y, int flags, void* userdata) {
  if (event == cv::EVENT_MOUSEMOVE) {
    std::cout << "Mouse Position: (" << x << ", " << y << ")" << std::endl;
  }
}

int main(int argc, char** argv) {
  if (argc < 2) {
    std::cerr << "Usage: " << argv[0] << " <image_path>\n";
    return 1;
  }
  std::string imagepath = argv[1];
  cv::Mat img = cv::imread(imagepath);
  if (img.empty()) {
    std::cerr << "Failed to load image.\n";
    return 1;
  }

  cv::namedWindow("image", cv::WINDOW_AUTOSIZE);
  cv::setMouseCallback("image", onMouse, nullptr);

  while (true) {
    cv::imshow("image", img);
    int key = cv::waitKey(10);
    if (key == 27 || key == 'q') break;
  }

  return 0;
}
